generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/relations/AfterOffsetRelation.java

```diff
@@ -24,51 +8,58 @@
 import com.scottlogic.deg.generator.profile.constraints.Constraint;
 import com.scottlogic.deg.generator.restrictions.linear.LinearRestrictions;
 
-public class AfterRelation<T extends Comparable<T>> implements FieldSpecRelations {
+public class AfterOffsetRelation<T extends Comparable<T>> implements FieldSpecRelations {
     private final Field main;
     private final Field other;
     private final boolean inclusive;
     private final LinearDefaults<T> defaults;
+    private final Granularity<T> offsetGranularity;
+    private final int offset;
 
-    public AfterRelation(Field main, Field other, boolean inclusive, LinearDefaults<T> defaults) {
+    public AfterOffsetRelation(Field main, Field other, boolean inclusive, LinearDefaults<T> defaults, Granularity<T> offsetGranularity, int offset) {
         this.main = main;
         this.other = other;
         this.inclusive = inclusive;
         this.defaults = defaults;
+        this.offsetGranularity = offsetGranularity;
+        this.offset = offset;
     }
 
     @Override
     public FieldSpec createModifierFromOtherFieldSpec(FieldSpec otherFieldSpec) {
-        if (otherFieldSpec instanceof NullOnlyFieldSpec){
+        if (otherFieldSpec instanceof NullOnlyFieldSpec) {
             return FieldSpecFactory.nullOnly();
         }
         if (otherFieldSpec instanceof WhitelistFieldSpec) {
             throw new UnsupportedOperationException("cannot combine sets with after relation, Issue #1489");
         }
 
-        LinearRestrictions<T> lr = (LinearRestrictions)((RestrictionsFieldSpec) otherFieldSpec).getRestrictions();
-        return createFieldSpec(lr.getMin(), lr.getGranularity());
+        LinearRestrictions<T> otherRestrictions = (LinearRestrictions) ((RestrictionsFieldSpec) otherFieldSpec).getRestrictions();
+        T min = otherRestrictions.getMin();
+        Granularity<T> granularity = offsetGranularity != null ? offsetGranularity : otherRestrictions.getGranularity();
```

Comment:
This felt a bit strange but as far as I can tell it's correct to use the granularity from `otherRestrictions` if there's none specified.